DSCResources/MSFT_NetworkTeam/MSFT_NetworkTeam.psm1

```diff
@@ -0,0 +1,243 @@
+
+        [ValidateSet('Present', 'Absent')]
+        [String]$ensure = 'Present'
+    )
+    Write-Verbose ($localizedData.GetTeamInfo -f $name)
+    $networkTeam = Get-NetLBFOTeam -Name $name -ErrorAction SilentlyContinue
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($networkTeam)
+        {
+            Write-Verbose ($localizedData.foundTeam -f $name)
+            $setArguments = @{
+                'name' = $name
+            }
+
+            if ($networkTeam.loadBalancingAlgorithm -ne $loadBalancingAlgorithm)
+            {
+                Write-Verbose ($localizedData.lbAlgoDifferent -f $loadBalancingAlgorithm)
+                $SetArguments.Add('loadBalancingAlgorithm', $loadBalancingAlgorithm)
+                $isNetModifyRequired = $true
+            }
+
+            if ($networkTeam.TeamingMode -ne $teamingMode)
+            {
+                Write-Verbose ($localizedData.teamingModeDifferent -f $teamingMode)
+                $setArguments.Add('teamingMode', $teamingMode)
+                $isNetModifyRequired = $true
+            }
+            
+            if ($isNetModifyRequired)
+            {
+                Write-Verbose ($localizedData.modifyTeam -f $name)
+                Set-NetLbfoTeam @setArguments -ErrorAction Stop -Confirm:$false
+            }
+
+            $netTeamMembers = Compare-Object -ReferenceObject $teamMembers -DifferenceObject $networkTeam.Members
+            if ($null -ne $netTeamMembers)
+            {
+                Write-Verbose ($localizedData.membersDifferent -f $name)
+                $membersToRemove = ($netTeamMembers | Where-Object {$_.SideIndicator -eq '=>'}).InputObject
+                if ($membersToRemove)
+                {
+                    Write-Verbose ($localizedData.removingMembers -f ($membersToRemove -join ','))
+                    $null = Remove-NetLbfoTeamMember -Name $membersToRemove -Team $name -ErrorAction Stop -Confirm:$false
+                }
+
+                $membersToAdd = ($netTeamMembers | Where-Object {$_.SideIndicator -eq '<='}).InputObject
+                if ($membersToAdd)
+                {
+                    Write-Verbose ($localizedData.addingMembers -f ($membersToAdd -join ','))
+                    $null = Add-NetLbfoTeamMember -Name $membersToAdd -Team $name -ErrorAction Stop -Confirm:$false
+                }
+            }
+            
+        } 
+        else 
+        {
+            Write-Verbose ($localizedData.createTeam -f $name)
+            $null = New-NetLbfoTeam -Name $name -TeamMembers $teamMembers -TeamingMode $teamingMode -LoadBalancingAlgorithm $loadBalancingAlgorithm -ErrorAction Stop -Confirm:$false
+            $timeout = 0
+            While ((Get-NetLbfoTeam -Name $name).status -ne 'Up')
+            {
+                Write-Verbose $localizedData.waitingForTeam
+                if ($timeout -ge 120)
+                {
+                    throw $localizedData.failedToCreateTeam   
```

Comment:
Instead of calling a throw here, can you use the following construct to throw the exception?
https://github.com/PowerShell/xNetworking/blob/dev/DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1#L708

I know this is a bit longer winded, but it creates a custom exception object that we can actually test for in unit tests.

So you'd end up with this:
```powershell
        $errorId = 'TeamCreateError'
        $errorCategory = [System.Management.Automation.ErrorCategory]::InvalidOperation
        $errorMessage = $localizedData.failedToCreateTeam
        $exception = New-Object -TypeName System.InvalidOperationException `
            -ArgumentList $errorMessage
        $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
            -ArgumentList $exception, $errorId, $errorCategory, $null

        $PSCmdlet.ThrowTerminatingError($errorRecord)
```